app/src/test/java/com/github/se/eventradar/EventOverviewViewModelTest.kt

```diff
@@ -131,7 +131,8 @@ class EventsOverviewViewModelTest {
 
     userRepository.addUser(mockUser)
     // MockUser is on the attendeeList for events with id "1" and "2"
-    viewModel.getUpcomingEvents("user1")
+    (userRepository as MockUserRepository).updateCurrentUserId("user1")
```

Comment:
Is that a reason why you casted this as MockUserRepository? I think it's been setup as `userRepository = MockUserRepository()` @Before the tests run, curious to hear your reason.

Reply:
Yes, the method updateCurrentUserID() is exclusive to the MockUserRepository and is not defined in the IUserRepository interface. As a result, casting is necessary to access this method, since it would not be available through the interface otherwise.